sendermode.go

```diff
@@ -154,36 +157,62 @@ func (fm *Frontman) sendResultsChanToHubQueue(resultsChan *chan Result) error {
 				shouldReturn = true
 			} else {
 				results = append(results, res)
-				if len(results) < fm.Config.QueueSenderBatchSize && len(*resultsChan) > 0 {
+				if len(results) < fm.Config.SenderBatchSize && len(*resultsChan) > 0 {
 					continue
 				}
 			}
 		}
 
-		duration := time.Since(lastSentToHub)
-		if duration >= interval {
-
-			if len(results) >= fm.Config.QueueSenderBatchSize {
-				sendResults = results[0:fm.Config.QueueSenderBatchSize]
-				results = results[fm.Config.QueueSenderBatchSize:]
+		if time.Since(lastSentToHub) >= sendInterval {
+			if len(results) >= fm.Config.SenderBatchSize {
+				sendResults = results[0:fm.Config.SenderBatchSize]
+				results = results[fm.Config.SenderBatchSize:]
 			} else {
 				sendResults = results
 				results = nil
 			}
 
 			if len(sendResults) > 0 {
-				logrus.Infof("Send %d results to hub, %d results in queue", len(sendResults), len(results))
 				lastSentToHub = time.Now()
 				go func(r []Result) {
 					err := fm.postResultsToHub(r)
 					if err != nil {
+						if err == ErrorHubGeneral {
+							// If the hub doesn't respond with 2XX, the results remain in the queue.
+							results = append(results, sendResults...)
+						}
 						logrus.Errorf("postResultsToHub error: %s", err.Error())
 					}
 
 					var m runtime.MemStats
 					runtime.ReadMemStats(&m)
-					logrus.Debugf("SenderModeQueue: Alloc = %v, TotalAlloc = %v, Sys = %v, NumGC = %v", m.Alloc/1024, m.TotalAlloc/1024, m.Sys/1024, m.NumGC)
+					logrus.Debugf("Sender queue: Alloc = %v, TotalAlloc = %v, Sys = %v, NumGC = %v", m.Alloc/1024, m.TotalAlloc/1024, m.Sys/1024, m.NumGC)
 				}(sendResults)
+			} else {
+				logrus.Debugf("Sender queue: nothing to do. outgoing queue empty.")
+			}
+		}
+
+		if time.Since(lastQueueStatsWrite) >= writeQueueStatsInterval {
+			lastQueueStatsWrite = time.Now()
+
+			data, _ := json.Marshal(map[string]int{
```

Comment:
error is ignored

Reply:
Yes, there is not much value added from this error. I think json.Marshal here can only fail if we ran out of memory.

Reply:
an sca would show an error here therefore the comment. thanks for resolving